src/main/resources/reference.conf

```diff
@@ -243,15 +243,15 @@ divolte {
     // the Kafka producer.
     // See: http://kafka.apache.org/documentation.html#producerconfigs
     producer = {
-      metadata.broker.list = ["localhost:9092"]
-      metadata.broker.list = ${?DIVOLTE_KAFKA_BROKER_LIST}
-
+      bootstrap.servers = ["localhost:9092"]
+      bootstrap.servers = ${?DIVOLTE_KAFKA_BROKER_LIST}
       client.id = divolte.collector
       client.id = ${?DIVOLTE_KAFKA_CLIENT_ID}
 
-      request.required.acks = 0
-      message.send.max.retries = 5
-      retry.backoff.ms = 200
+      acks = 1
+      retries = 0
+      compression.type = lz4
+      max.in.flight.requests.per.connection = 1
```

Comment:
These are defaults. Users can set them to anything else in their own deployments.

More than 1 in flight per connection implies the possibility of re-ordering in case of retry logic (in the case of multiple in-flights, it's undefined which error reaches our code first). We'll leave it to the user whether this is acceptable.

With `block.on.buffer.full` false, sending will result in a immediate exception when sending. The resulting exception is not retriable (`BufferExhaustedException` directly extends `KafkaException`, not `RetriableException`), meaning we'd immediately drop the event at that point instead of keeping it queued for a retry. The send buffer can be full for many reasons, some recoverable, some not. In either case, once messages made it to the send buffer, it's out of our control and in case of a unrecoverable failure on Kafka's side, we'll lose everything in the send buffer (but not everything in our own queue), without knowing whether it was sent or not. You can try to defeat this with retries being handled by the Kafka client automatically.

In short: the Kafka client attempts to do everything we need in terms of internal queueing and async usage, however it will in the end still block and throw exceptions nonetheless in the presence of distributed failures. In order to make sure that this does not interfere with other sinks (HDFS), we separate it out to a different thread with a queue in between.